src/Objects/_PVector.js

```diff
@@ -0,0 +1,283 @@
+             len === 2? _PVector.dot(v, y) :
+             len > 2? this.x*v + this.y*y + this.z*z :
+             argsErr('dot', len, 1);
+    },
+    cross: function(v1, v2, target) {
+      return target? _PVector.cross(v1, v2, target)
+                   : _PVector.cross(this, v1, v2);
+    },
+    lerp: function() {
+      var x, y, z, amt;
+      const len = arguments.length;
+      if ((len | 1) === 1) { argsErr('lerp', len, 2); }
+      if (len === 2) { // given vector and amt
+        const v = arguments[0];
+        x = v.x;
+        y = v.y;
+        z = v.z;
+        amt = arguments[1];
+      } else if (len === 3) { // given vector 1, vector 2 and amt
+        return _PVector.lerp(arguments[0], arguments[1], arguments[2]);
+      } else { // given x, y, z and amt
+        x = arguments[0];
+        y = arguments[1];
+        z = arguments[2];
+        amt = arguments[3];
+      }
+      this.x = lerp(this.x, x, amt);
+      this.y = lerp(this.y, y, amt);
+      this.z = lerp(this.z, z, amt);
+      return this;
+    },
+    add: function(v, y, z) {
+      if (y instanceof _PVector) { return _PVector.add(v, y, z); }
+      const len = arguments.length;
+      if (len === 1) {
+        //_PVector.add(this, v, this);
+        this.x += v.x;
+        this.y += v.y;
+        this.z += v.z;
+      } else if (len > 1) {
+        this.x += v;
+        this.y += y;
+        if (len > 2) { this.z += z };
+      } else { argsErr('add', len, 1); }
+      return this;
+    },
+    sub: function(v, y, z) {
+      if (y instanceof _PVector) { return _PVector.sub(v, y, z); }
+      const len = arguments.length;
+      if (len === 1) {
+        this.x -= v.x;
+        this.y -= v.y;
+        this.z -= v.z;
+      } else if (len > 1) {
+        this.x -= v;
+        this.y -= y;
+        if (len > 2) { this.z -= z; }
+      } else { argsErr('sub', len, 1); }
+      return this;
+    },
+    mult: function(v, n, target) {
+      const len = arguments.length;
+      if (len === 1) {
+        if (typeof v === 'number') {
+          this.x *= v;
+          this.y *= v;
+          this.z *= v;
+        } else {
+          this.x *= v.x;
+          this.y *= v.y;
+          this.z *= v.z;
+        }
+        return this;
+      } else if (len > 1) { return _PVector.mult(v, n, target); }
+      argsErr('mult', len, 1);
+    },
+    div: function(v, n, target) {
+      const len = arguments.length;
+      if (len === 1) {
+        if (typeof v === 'number') {
+          this.x /= v;
+          this.y /= v;
+          this.z /= v;
+        } else {
+          this.x /= v.x;
+          this.y /= v.y;
+          this.z /= v.z;
+        }
+        return this;
+      } else if (len > 1) { return _PVector.div(v, n, target); }
+      argsErr('div', len, 1);
+    },
+    toString: function() {
+      return '[' + this.x + ', ' + this.y + ', ' + this.z + ']';
+    },
+    equals: function(o) {
+      return o === this? true : o instanceof _PVector?
```

Comment:
wow, that's silly of me, the 2nd case is supposed to be "if v1.eq1, use eq2". I've updated the test as http://jsperf.com/nested-ternary-equality/4 and with the use of `eq2` in the second case it's now *wildly* more efficient without the ternaries.

(There's also an odd warning about "P(...)[0] is undefined.", but that warning only shows up in jsperf, not if the code is run anywhere else like jsbin or jsfiddle, so I suspect it's related to something jsbin does rather than this code specifically)